core/src/main/java/org/apache/iceberg/RemoveSnapshots.java

```diff
@@ -190,6 +301,10 @@ public void commit() {
         .onlyRetryOn(CommitFailedException.class)
         .run(item -> {
           TableMetadata updated = internalApply();
+          if (cleanExpiredFiles && updated.refs().size() > 1) {
+            throw new UnsupportedOperationException("Cannot incrementally clean files for tables with more than 1 ref");
```

Comment:
@amogh-jahagirdar  I have a simple use case blocked by this line. 

I am tagging one of the snapshots on the mainline as *never expire*, but as soon as I tag it, the normal expire procedure (expireOlderThan) stopped working, as there are two refs now.

is the constraint correct here ? because in my scenario, I am merely indicating SKIP this snapshot, with other terms stay the same.

Reply:
I think, if there's only one branch main, and tags are all on main, it is safe to carry on. 
is that correct ?